docs/source/toolkit/testing.rst

```diff
@@ -0,0 +1,213 @@
+:code:`globus_action_provider_tools.testing.fixtures*`:
+
+.. code-block:: python
+                
+    from globus_action_provider_tools.testing.fixtures import (
+        flask_helpers_noauth
+    )
+
+Once imported, simply pass the :code:`flask_helpers_noauth` fixture as a
+parameter to another fixture which creates the app and returns a resource for
+use in tests. Unlike the :code:`apt_blueprint_noauth` fixture, the
+:code:`flask_helpers_noauth` fixture does not need to be explicitly executed -
+simply passing it as a parameter is sufficient to temporarily stub out the
+Provider's authentication. An example of how to create a :code:`client` fixture
+which can be used to make unauthenticated HTTP requests against the Action
+Provider is shown below:
+
+.. code-block:: python
+
+    @pytest.fixture(scope="module")
+    def client(flask_helpers_noauth):
+        app = create_app()
+        app.config["TESTING"] = True
+        yield app.test_client()
+
+Once composed like this, you can use the :code:`client` fixture in your tests to
+receive and use a Flask *test_client* to make requests against your Action
+Provider:
+
+.. code-block:: python
+
+    def test_introspection_endpoint(client):
+        response = client.get("/")
+        assert response.status_code == 200
+
+.. note:: 
+    
+    The :code:`flask_helpers_noauth` fixture will patch the TokenChecker in a
+    global scope during testing, meaning that any other Action Providers that
+    are themselves built using the Flask API Helpers will also have their
+    TokenChecker's patched. This may lead to unintended issues if testing
+    multiple Action Providers in the same pytest test session. If this is your
+    case, we highly recommend isolating your Action Provider tests.
+
+
+Mocks
+^^^^^
+
+The toolkit provides various `mocks
+<https://docs.python.org/3/library/unittest.mock.html#the-mock-class>`_ which
+can be used individually to stub out your Action Provider's authentication. You
+should use these directly if you are writing an Action Provider using a
+non-Flask framework or if you've decided not to use the built in Flask helpers.
+
+.. note::
+
+    This toolkit uses these mocks within the
+    :code:`globus_action_provider_tools.testing.fixtures` module.  
+
+
+.. _mock-authstate:
+
+Mock AuthState
+--------------
+
+An *AuthState* represents a requestor's authentication status and Globus Auth
+information. Every request should have its token validated via the
+*TokenChecker*'s :code:`check_token` method, which in turns generates an
+*AuthState* object.
+
+During testing, it is desirable to not need to provide valid tokens with every
+request. Use the :code:`mock_authstate` mock to generate a stubbed out
+*AuthState* object that won't validate requestor properties against Globus Auth.
+This is most useful when used in a patch as the return value for the
+*TokenChecker*'s :code:`check_token` method:
+
+.. code-block:: python
+
+    from unittest import mock     
```

Comment:
If we're using pytest, does using its monkeypatch module make at least as much sense as using `unittest.mock`? Do we have concerns about mixing unittest and pytest?

Reply:
I think the only concern is providing a uniform non-confusing story for folks who are ramping up with testing/using the tools. `unittest` and `pytest` play pretty well with each other so I'm not too worried about them interfering with each other (unless there are cases and I just haven't run across them yet). 

There are some gotchas with using pytest's `monkeypatch` fixture because it's function-scoped. The fixtures provided here are using `patch` and `mock` directly so they can be used in fixtures regardless of their scope. I like the idea of using `monkeypatch` in the examples since it's syntax is clearer than patch's.